pubspec.yaml

```diff
@@ -8,19 +8,19 @@ environment:
   sdk: ">=2.18.0 <4.0.0"
 
 dependencies:
-  build: ^3.0.0
-  build_config: ^1.0.0
-  characters: ^1.2.0
-  collection: ^1.15.0
-  dart_style: ^3.0.0
-  diacritic: ^0.1.3
-  file: ^7.0.0
-  meta: ^1.7.0
-  path: ^1.8.0
-  yaml: ^3.1.0
+  build: ^4.0.0
+  build_config: ^1.2.0
+  characters: ^1.4.0
+  collection: ^1.19.1
+  dart_style: ^3.1.2
+  diacritic: ^0.1.6
+  file: ^7.0.1
+  meta: ^1.16.0
+  path: ^1.9.1
+  yaml: ^3.1.3
```

Comment:
For business apps, I’d say it makes sense to stick with the latest dependencies and lock versions. For SDKs, though, I’d prefer to keep things as flexible as possible for users, so they’re not forced to upgrade unless we hit an actual issue with a transitive dependency that requires updating the constraint.

In this case, `build_config` sets the minimum SDK to 3.7.0, which is higher than the plugin’s minimum SDK. I haven't checked all the changelogs, maybe there are more changes like that. Most likely no one would notice, and we _could_ just upgrade. But since we’re not using any new features from these dependencies and the plugin should work fine with older versions, I’d suggest only bumping the `build` version.

What do you think?

Reply:
Oh, wait, `build` itself is constrained to `3.7.0`. Let me check what that means for the package.

Reply:
Yeap, please revert all unnecessary upgrades to constraints to all `dependencies`. I'll deal with warnings before publishing a new version.